optuna/testing/sampler.py

```diff
@@ -0,0 +1,32 @@
+import optuna
+
+if optuna.types.TYPE_CHECKING:
+    from typing import Dict  # NOQA
+
+    from optuna.distributions import BaseDistribution  # NOQA
+    from optuna.structs import FrozenTrial  # NOQA
+    from optuna.study import InTrialStudy  # NOQA
+
+
+class DeterministicRelativeSampler(optuna.samplers.BaseSampler):
+    def __init__(self, relative_search_space, relative_params):
+        # type: (Dict[str, BaseDistribution], Dict[str, float]) -> None
+
+        self.relative_search_space = relative_search_space
+        self.relative_params = relative_params
+
+    def infer_relative_search_space(self, study, trial):
+        # type: (InTrialStudy, FrozenTrial) -> Dict[str, BaseDistribution]
+
+        return self.relative_search_space
+
+    def sample_relative(self, study, trial, search_space):
+        # type: (InTrialStudy, FrozenTrial, Dict[str, BaseDistribution]) -> Dict[str, float]
+
+        return self.relative_params
+
+    def sample_independent(self, study, trial, param_name, param_distribution):
+        # type: (InTrialStudy, FrozenTrial, str, BaseDistribution) -> float
+
+        sampler = optuna.samplers.RandomSampler()
```

Comment:
IMO, the use of `RandomSampler` in this class is a bit confusing because deterministic behavior is expected from the name. Do you have a specific reason to use `RandomSampler`?

Reply:
No. We can change the code to not use `RandomSampler`.

Reply:
Fixed: https://github.com/pfnet/optuna/pull/380/commits/0d58d4e77dfd688f016f37030ff9400d305af063
Thank you for your comment.